universal-login-react/src/ui/Onboarding/Onboarding.tsx

```diff
@@ -54,11 +56,31 @@ export const Onboarding = (props: OnboardingProps) => {
               </div>
             </OnboardingStepsWrapper>}
         />
+        <Route
+          path='/email'
+          exact
+          render={({history}) =>
+            <EmailFlowChooserScreen
+              walletService={props.walletService}
+              hideModal={props.hideModal}
+              onConnectClick={() => console.log('connect not supported yet!')}
+              onCreateClick={async (email, ensName) => {const requestPromise = props.walletService.createRequestedWallet(email, ensName); history.push('/code'); await requestPromise;}}
```

Comment:
This code will have the same result I guess:

```suggestion
              onCreateClick={async (email, ensName) => {
              props.walletService.createRequestedWallet(email, ensName);
              history.push('/code')}}
```

Reply:
What if this promise will be rejected? Will we have error: unhandled promise rejection?

Reply:
Yes, we don't handle it, but:
How do we handle an error in the case, when we wait for the promise but initiate function by synchronous click?

Reply:
Don't we have some kind of error boundary?

Reply:
Error boundary works well for sync errors, so if you want use error boundary you should wrap async function with ```try {...} catch () {...}```